### DISCUSSION TITLE
After ^13.4.10 - layout.tsx will return server errors if you specify paths like (website) and (admin) inside app folder

### DISCUSSION THREAD

**Discussion by Galanthus** [vercel/next.js#52721] [App Router]
After 4 hours of debugging. I have found a bug inside Next ^13.4.10. 

My current folder structure:
app: 
 - (website)
 - - layout.tsx (works in previous version)
 - (admin)

This will work:
- app
- - layout.tsx

But, after upgrading layout.tsx will return server errors if it's inside (website), you have to move it outside and in the root of "app".

Global styles won't load neither. 

error:

```
^[Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: null.
Error: Unsupported Server Component type: Null
    at attemptResolveElement (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1421:19)
    at resolveModelToJSON (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1627:41)
    at Object.toJSON (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1169:28)
    at stringify (<anonymous>)
    at processModelChunk (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:2116:24)
    at retryTask (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1882:38)
    at performWork (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1920:21)
    at Timeout.eval [as _onTimeout] (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1429:28)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7) {
  digest: '135229182'
}
```
Can someone confirm please?

**Comment by theDanielJLewis** [comment]
This must be what's causing the the `Error: Unsupported Server Component type: Null` and `Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.` for me, too.

I use an organization structure like`(website)/layout.tsx` and `(admin)/layout.tsx`, too, since that's the only way to fully overwrite a layout for different groups (for example, having different sidebars).

**Reply by nass59** [reply]
Same issue for me with same multiple root layouts. I tried also the 13.4.11-canary.0 but same error

**Comment by theDanielJLewis** [comment]
Are you certain it's the folder issue, and not what you also reported in #52699?

**Reply by Galanthus** [reply]
It is 100% latest Next js. Tried with a new clean project.

**Reply by theDanielJLewis** [reply]
I mean is it only this `(website)` folder structure in 13.4.10 for sure? Or might the XML problem you reported indicate an additional problem?

**Reply by Galanthus** [reply]
It is not XML i have debugged for 4 hours to figure this out. You can create a new project, add (website) inside folder layout [slug]/page.tsx fetch random users, and add layout.tsx inside that folder and run NotFound().

**Comment by emmgfx** [comment]
I'm having a similar issue after upgrading to 13.4.10, still debugging.

**Comment by icyJoseph** [comment]
There's a reproduction demo already, I think you can go ahead and create an issue: https://stackblitz.com/edit/stackblitz-starters-mtdjqq?file=app%2F[locale]%2Fabout%2F[slug]%2Fpage.js visit /en/about/test and boom

**Reply by icyJoseph** [reply]
Although in this case the problem is the missing root layout.

**Comment by icyJoseph** [comment]
Could the problem be not having a root layout? like if a path ends up not having a top level parent layout, it crashes?

**Reply by Galanthus** [reply]
No. It is an issue in the latest Next version. Didn’t have any issues before

**Reply by icyJoseph** [reply]
Yeah, but do you have a root layout for all possible routes? It seems like that's the easiest way to reproduce the error, but it might not be the same cause.

Perhaps a root layout for all routes is always required? https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required And maybe now it is enforcing it?

**Reply by Galanthus** [reply]
Root is inside alias (website) and another in (admin) never had one in app folder directly with no issues.

**Reply by icyJoseph** [reply]
Right, that's why I think you've found a different way to trigger this, but since the only reproduction demo https://stackblitz.com/edit/stackblitz-starters-mtdjqq?file=app%2F[locale]%2Fabout%2F[slug]%2Fpage.js I have is this one I can't conclude anything :( 

Could you please try to make one? you can use the above as base, and just share the link to the forked version, just do `ctrl+s` and it'll fork a new one, and you can try to reproduce it there

**Reply by icyJoseph** [reply]
> You can use [route groups](https://nextjs.org/docs/app/building-your-application/routing/route-groups) to create multiple root layouts. See an [example here](https://nextjs.org/docs/app/building-your-application/routing/route-groups#creating-multiple-root-layouts).

And if we follow the link:

> ### [Creating multiple root layouts](https://nextjs.org/docs/app/building-your-application/routing/route-groups#creating-multiple-root-layouts)
> To create multiple [root layouts](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required), remove the top-level layout.js file, and add a layout.js file inside each route groups.


It says to add a layout group, "inside each route group".

**Reply by Galanthus** [reply]
Yes I know that but still problem persist if you call a notFound function the console will output errors.

**Comment by theDanielJLewis** [comment]
This seems to be fixed in 13.4.12. Have you observed the same?

**Reply by Galanthus** [reply]
Not yet perfect. It doesn't return the custom not-found.tsx page but instead displays the default 404.

```
  if (!data) {
    notFound()
  }
```

**Reply by theDanielJLewis** [reply]
I was having that problem before 13.4.10.

**Reply by Galanthus** [reply]
Works perfectly fine in 13.4.9 reverted back shows my custom 404 but updating to 13.4.10 and newer does not.

**Reply by emmgfx** [reply]
In `13.4.12` I don't have the error, but isn't working well.

Looks like it's obviating the layout file and the middleware. And I need booth because I instantiate the i18n context in the layout and rewrite urls to hide the default locale path, with the middleware.
